CMakeLists.txt

```diff
@@ -329,6 +329,7 @@ endif()
 # Pull in our subdirectories
 add_subdirectory(src)
 add_subdirectory(examples)
+add_subdirectory(tests)
```

Comment:
I think you should push the `tests` directory under `src`. It makes some sense for the `examples` code to live at the root, since in some way those programs are clients of the library, not part of it. But I'd consider the tests to be "part of" the SDK. Fewer top level directories containing source code is generally a good idea.